### PR TITLE
enable code linting check for all supported Python versions

### PR DESCRIPTION
**Pull request by JensTimmerman**
run linting tests for all supported python versions

### CODE REVIEW

**Comment by JensTimmerman** [comment]
seems to work, tests fail because we're now catching errors:

> ./easybuild/tools/configobj.py:2026:55: F812 list comprehension redefines 'line' from line 2014

**Comment by akesandgren** [comment]
@boegel This looks ok to me, and the caught error might need to be fixed ....

**Comment by akesandgren** [comment]
@JensTimmerman could you do "eb --sync-pr-with-develop 3725 --pr-target-repo easybuild-framework" on this one to make sure it's up to date.

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -18,4 +22,4 @@ jobs:
 
     - name: Run flake8 to verify PEP8-compliance of Python code
       run: |
-        flake8
+        flake8 --exclude ./easybuild/tools/py2vs3/py3.py
```
[the remark this answers is not in the document]
Comment (boegel) [reply .github/workflows/linting.yml]: fixed in a2ec58b8a

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Thanks a lot @JensTimmerman!

**Pull request merged**